ocaml/libs/xapi-stdext/lib/xapi-stdext-threads/threadext.ml

```diff
@@ -131,3 +131,17 @@ let wait_timed_write fd timeout =
       true
   | _ ->
       assert false
+
+module Pthread = struct
+  external c_set_name : string -> unit = "stub_set_name"
+
+  let set_name s =
+    let len = Int.min (String.length s) 15 in
+    let tname = String.sub s 0 len in
+    try
+      let () = c_set_name tname in
+      Some tname
+    with _ -> None
```

Comment:
I prefer the old code. Why worrying throwing an exception just to then ignore it? Either handle or at least log somehow or ignore in the C stub.